plugins/modules/inventory_intent.py

```diff
@@ -2512,62 +2513,104 @@ def get_diff_merged(self, config):
                 self.check_return_status()
                 csv_reader = self.decrypt_and_read_csv(export_response, password)
                 self.check_return_status()
-                device_data = next(csv_reader, None)
-                playbook_params = self.want.get("device_params")
-                playbook_params['ipAddress'] = device_to_update
-
-                if not playbook_params['cliTransport']:
-                    playbook_params['cliTransport'] = device_data['protocol']
-                if not playbook_params['snmpPrivProtocol']:
-                    playbook_params['snmpPrivProtocol'] = device_data['snmpv3_privacy_type']
-
-                csv_data_dict = {
-                    'username': device_data['cli_username'],
-                    'password': device_data['cli_password'],
-                    'enable_password': device_data['cli_enable_password'],
-                }
+                device_details = {}
```

Comment:
Can you please take this as an enhancement and do it when you get available time?

This API has lot more functionalities. Can we split it into multiple APIs?

Also the following code is repeated for 3 times. We can use the API "    def check_execution_response_status(self, response):" in dnac.py. 

                            if response and isinstance(response, dict):
                                    task_id = response.get('response').get('taskId')

                                    while True:
                                        execution_details = self.get_task_details(task_id)
                                        if execution_details.get("isError"):
                                            self.status = "failed"
                                            failure_reason = execution_details.get("failureReason")
                                            if failure_reason:
                                                self.msg = """Device new management IP updation for device '{0}'
                                                         get failed due to {1}""".format(device_ip, failure_reason)
                                            else:
                                                self.msg = "Device new management IP updation for device '{0}' get failed".format(device_ip)
                                            self.log(self.msg, "ERROR")
                                            break
                                        elif execution_details.get("endTime"):
                                            self.status = "success"
                                            self.result['changed'] = True
                                            self.result['response'] = execution_details
                                            self.msg = """Device '{0}' present in Cisco Catalyst Center and new management ip '{1}' have been
                                                     updated successfully""".format(device_ip, new_mgmt_ipaddress)
                                            self.log(self.msg, "INFO")
                                            break

Reply:
addressed.